src/v/pandaproxy/rest/handlers.cc

```diff
@@ -72,6 +73,47 @@ ss::shard_id consumer_shard(const kafka::group_id& g_id) {
 
 ss::future<server::reply_t>
 get_brokers(server::request_t rq, server::reply_t rp) {
+    ss::sstring username{};
+    ss::sstring userpass{};
+    if (rq.service().config().rest_require_auth()) {
+        auto creds_pair = parse::basic_auth_header(*rq.req);
+        username = creds_pair.first;
+        userpass = creds_pair.second;
+
+        // Create new kafka client config with scram creds
+        // and use sha256 by default
+        auto client_cfg = std::move(rq.service().client_config());
+        client_cfg.sasl_mechanism.set_value(ss::sstring{"SCRAM-SHA-256"});
+        client_cfg.scram_username.set_value(username);
+        client_cfg.scram_password.set_value(userpass);
+
+        // Restart kafka client with new configs
+        auto client_cfg_as_yml = to_yaml(
+          client_cfg, config::redact_secrets::no);
+        co_await rq.service().client().stop().then_wrapped(
+          [&rq, &client_cfg_as_yml, &username](ss::future<> f) {
+              if (f.failed()) {
+                  throw std::runtime_error("Failed to stop proxy client");
+              }
+
+              // TODO(@NyaliaLui): Check if the lambda in
+              // application.h::construct_service is still referring to this
+              // same client after we stop and restart it.
+              return rq.service()
+                .client()
+                .start(client_cfg_as_yml)
```

Comment:
Starting and stopping a service while the request is serviced seems like a major hit to latency. I should find a better place to do this.